python/tvm/relay/transform/quantization/quantized_operators/dense.py

```diff
@@ -0,0 +1,183 @@
+from typing import *
+
+import numpy as np
+import tvm
+from tvm import relay
+from tvm.relay.dataflow_pattern import (
+    DFPatternCallback,
+    _DFPatternCallback,
+    is_constant,
+    is_op,
+    wildcard,
+)
+from tvm.relay.op import nn, tensor
+from tvm.relay.transform.quantization import calibration_rewrite_utils
+
+QParams = NamedTuple(
+    "QParams", [("scale_factor", tvm.relay.Expr), ("zero_point", tvm.relay.Expr), ("dtype", str)]
+)
+
+
+class AffineQuantizationVarCreator:
+    """Class which manages references to our qparams and can insert state."""
+
+    def __init__(self):
+        self.ref_count = 0
+        self.qparams = []
+
+    def get_qparams(self, name_hint: str, dtype: str = "int8") -> QParams:
```

Comment:
Passing in a name to this function is good, it will help avoid confusion. 

Right now, your implementation saves every QParam ever created and adds it to a list. I'm not sure this is the exact behavior we want, since the purpose of the class saving the QParams is to let the C++ rewriter see the QParams that were created in the rewrite of the last pattern, not all the QParams. (We can get all the scales and zero point variables by doing relay.analysis.free_vars, so we don't have to store them all in a list). 

However, I'm not 100% sure of the implementation details of that because it will have to interact with the FFI. For now, maybe we should just not save the QParams we've previously created, and we can add the functionality when we need it.